### PR TITLE
Code Quality: Added more null ref protections when renaming items

### PR DESCRIPTION
**Pull request by yaira2**
Added more null ref protections when renaming items

### CODE REVIEW

File: src/Files.App/Views/Layouts/GridLayoutPage.xaml.cs [hunk 1]
```diff
@@ -183,27 +183,35 @@ override public void StartRenameItem()
...
+			if (FolderSettings?.LayoutMode == FolderLayoutModes.GridView)
 			{
-				Popup popup = gridViewItem.FindDescendant("EditPopup") as Popup;
-				TextBlock textBlock = gridViewItem.FindDescendant("ItemName") as TextBlock;
+				if (gridViewItem.FindDescendant("EditPopup") is not Popup popup)
+					return;
+
 				textBox = popup.Child as TextBox;
+				if (textBox is null)
+					return;
```
Comment (0x5bfa) [inline src/Files.App/Views/Layouts/GridLayoutPage.xaml.cs:202]:
```suggestion
				if (popup.Child is not TextBox textBox)
					return;
```

File: src/Files.App/Views/Layouts/GridLayoutPage.xaml.cs [hunk 2]
```diff
@@ -183,27 +183,35 @@ override public void StartRenameItem()
...
 				textBlock.Opacity = 0;
 				popup.IsOpen = true;
 				OldItemName = textBlock.Text;
 			}
 			else
 			{
-				TextBlock textBlock = gridViewItem.FindDescendant("ItemName") as TextBlock;
 				textBox = gridViewItem.FindDescendant("TileViewTextBoxItemName") as TextBox;
+				if (textBox is null)
+					return;
```
Comment (0x5bfa) [inline src/Files.App/Views/Layouts/GridLayoutPage.xaml.cs:213]:
```suggestion
				if (gridViewItem.FindDescendant("TileViewTextBoxItemName") is not TextBox textBox)
					return;
```

**Review by 0x5bfa — COMMENTED** [review]
Simplified type checks

File: src/Files.App/Views/Layouts/GridLayoutPage.xaml.cs (hunk 2 above)
Comment (yaira2) [reply src/Files.App/Views/Layouts/GridLayoutPage.xaml.cs:213]: `textBox` is already defined on 192 so I don't think this will work

**Review by hishitetsu — APPROVED** [review]
LGTM

**Pull request merged**